spec/controllers/mixins/ems_common_angular_spec.rb

```diff
@@ -0,0 +1,27 @@
+describe Mixins::EmsCommonAngular do
+  context '.retrieve_event_stream_selection' do
+    let(:network_controller) { EmsNetworkController.new }
+    let(:ems_nuage) { FactoryGirl.create(:ems_nuage_network_with_authentication) }
+
+    it 'when amqp' do
+      ems_nuage.endpoints << Endpoint.create(:role => 'amqp')
+      network_controller.instance_variable_set(:@ems, ems_nuage)
+
+      expect(network_controller.send(:retrieve_event_stream_selection)).to eq('amqp')
+    end
+
+    it 'when ceilometer' do
+      ems_openstack = FactoryGirl.create(:ems_openstack_with_authentication)
+      ems_openstack.endpoints << Endpoint.create(:role => 'ceilometer')
+      network_controller.instance_variable_set(:@ems, ems_openstack)
+
+      expect(network_controller.send(:retrieve_event_stream_selection)).to eq('ceilometer')
+    end
+
+    it 'when none' do
+      network_controller.instance_variable_set(:@ems, ems_nuage)
+
+      expect(network_controller.send(:retrieve_event_stream_selection)).to eq('none')
+    end
+  end
+end
```

Comment:
Can you please add also following tests:

* AMQP when ceilometer has empty hostname
* CEILOMETER when amqp has empty hostname
* NONE when amqp and ceilometer have nil endpoints
* NONE when amqp and cailometer have empty hostnames